plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -2834,12 +2834,16 @@ public void dropNotNullConstraint(ConnectorSession session, ConnectorTableHandle
     }
 
     @Override
-    public TableStatisticsMetadata getStatisticsCollectionMetadataForWrite(ConnectorSession session, ConnectorTableMetadata tableMetadata)
+    public TableStatisticsMetadata getStatisticsCollectionMetadataForWrite(ConnectorSession session, ConnectorTableMetadata tableMetadata, boolean replace)
     {
         if (!isExtendedStatisticsEnabled(session) || !isCollectExtendedStatisticsOnWrite(session)) {
             return TableStatisticsMetadata.empty();
         }
 
+        if (replace) {
+            return getStatisticsCollectionMetadata(tableMetadata, Optional.empty(), _ -> {});
```

Comment:
keep it consistent with the 3 other  occurrences of `return getStatisticsCollectionMetadata(tableMetadata, Optional.empty(), availableColumnNames -> {});` below

```suggestion
            return getStatisticsCollectionMetadata(tableMetadata, Optional.empty(), availableColumnNames -> {});
```